### DISCUSSION TITLE
WSL not updated to 1.1.0

### DISCUSSION THREAD

**Discussion by thucke** [microsoft/WSL#9716] [General] [answered]
### Version

Microsoft Windows 10 [Version 21H2 Build 19044.2251]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.16

### Distro Version

SUSE Tumbleweed

### Repro Steps

* Without access to Microsoft Store MSI-Package has been downloaded from Github.
* Install in Powershell Admin-Box: Add-AppPackage -Path ".\Microsoft.WSL_1.1.0.0_x64_ARM64.msixbundle"

### Expected Behavior

* wsl --version shows detailed version info
* Kernel is updated to version v5.15.83
* systemd is operating after is has been activated in wsl.conf 

### Actual Behavior

* wsl --version shows help page
* wsl --status still shows "Kernel-Version: 5.10.16"
* systemd in distro still doesn't work (wsl.conf has been already updated)

**Comment by olivierchatry** [comment]
You are using a arm64 packages. Is your computer ARM ? or is it x64 ?

**Comment by sbradnick** [comment]
> You are using a arm64 packages. Is your computer ARM ? or is it x64 ?

That's a .msixbundle, so it contains both and will install the appropriate one.

In my experience, if you currently have one version installed and try to install an update - it fails until you remove the _old_ version ~ are you sure that's not happening here?

There are also pre-requisites for it to work in Win10, not sure if you've met them.

**Comment by elsaco** [comment]
The command is `Add-AppxPackage` not `Add-AppPackage`

**Comment by sbradnick** [comment]
If I don't remove v1.0.3, I can't install v1.1.0

![2023-01-27-125949_1600x967_scrot](https://user-images.githubusercontent.com/84082961/215160572-b277f4c4-5904-4232-a31c-b1005c1f946f.png)

Also, `wsl --status` & `wsl --version` should return results vs. a dump of `wsl --help` if things are updated correctly.

Latest Win10 updates (continue, as v1.0.3 did) to work for me w/ v1.1.0.

![2023-01-27-130634_1600x967_scrot](https://user-images.githubusercontent.com/84082961/215161829-b3ae1b65-f1c6-4966-ab57-478bfc96990a.png)

**Comment by thucke** [comment]
> The command is `Add-AppxPackage` not `Add-AppPackage`
Sorry - it was a typo - I useed the correct statement.

**Comment by thucke** [comment]
> If I don't remove v1.0.3, I can't install v1.1.0
> 
> ![2023-01-27-125949_1600x967_scrot](https://user-images.githubusercontent.com/84082961/215160572-b277f4c4-5904-4232-a31c-b1005c1f946f.png)
> 
> Also, `wsl --status` & `wsl --version` should return results vs. a dump of `wsl --help` if things are updated correctly.
> 
> Latest Win10 updates (continue, as v1.0.3 did) to work for me w/ v1.1.0.
> 
> ![2023-01-27-130634_1600x967_scrot](https://user-images.githubusercontent.com/84082961/215161829-b3ae1b65-f1c6-4966-ab57-478bfc96990a.png)

I experienced exacly that behaviour.
That's why I also deinstalled v1.0.3 incl. reboot before I installed v1.1.0.

I am experiencing the same behaviour on two machines I own (Notebook and Workstation).

Possibly important sidenote:
Both systems are managed by a foreign vendor of my company who disallowed direct access to MS Store.
That's why I need to install the appx-package.
Maybe there are addidinal GPO set which could influence the behaviour of the WSL installation?

**Comment by olivierchatry** [comment]
Maybe by forcing the upgrade / downgrade : 
```
 Add-AppxPackage -Path .\Microsoft.WSL_1.1.0.0_x64_ARM64.msixbundle -ForceUpdateFromAnyVersion  -ForceApplicationShutdown
```

Between, be careful with 1.1.0, there is a nasty issue with localhost networking. ( so might be better to wait for a new version ).

**Comment by thucke** [comment]
> Maybe by forcing the upgrade / downgrade :
> 
> ```
>  Add-AppxPackage -Path .\Microsoft.WSL_1.1.0.0_x64_ARM64.msixbundle -ForceUpdateFromAnyVersion  -ForceApplicationShutdown
> ```
> 
> Between, be careful with 1.1.0, there is a nasty issue with localhost networking. ( so might be better to wait for a new version ).
Thank you @olivierchatry ,
I meant that I've already tried both Force-Options but I tried it again.
Unfortunately also without any change even after reboot.

**Comment by sbradnick** [comment]
Can you provide screenshots of these commands being run and their output? If they're installing successfully there really shouldn't be any issue.

* Can you update Win10 to 22H2?
* What's the output of `wsl --status` / `wsl --version` (esp. if v1.0.3 || v1.1.0 install successfully)?
* I'm not sure how you listed `5.10.16[.3]` [kernel](https://learn.microsoft.com/en-us/windows/wsl/kernel-release-notes) in use for WSL, is that a typo? It's not that that kernel wasn't ever in use, but I'm not sure it's tied to any modern-ish WSL release.

**Comment by thucke** [comment]
Thank you @sbradnick ,
first - here are screenshots of the commands you asked for:
![2023-01-31 09_10_19-Clipboard](https://user-images.githubusercontent.com/15112795/215703237-95a1c636-8cfe-4ed7-b2a5-c19a08847f8b.png)
![2023-01-31 09_10_54-Clipboard](https://user-images.githubusercontent.com/15112795/215703259-52eb19cf-b7d8-4b64-b2ed-8957284c5ed0.png)

The windows version I've got provided by our vendor:
![2023-01-31 09_06_44-Clipboard](https://user-images.githubusercontent.com/15112795/215703315-95aa24d7-0dd8-4dd2-9bb2-7c3b4725cdb0.png)
Unfortunately it is not possible to update to 22H2 on my own.

It seems that the package has been installed correctly:
![2023-01-31 09_02_39-Clipboard](https://user-images.githubusercontent.com/15112795/215703549-61fc580b-084e-4ef6-ad32-c2d234bb6e6e.png)

but it seems also that the installed files were not set active - I see only onld version of the exe in system path:
![2023-01-31 09_05_29-Clipboard](https://user-images.githubusercontent.com/15112795/215703705-3a85b1a6-efcb-42e8-87c1-231545f7afc9.png)

and the new W10 version is not compatible yet: 
![2023-01-31 09_03_37-Clipboard](https://user-images.githubusercontent.com/15112795/215703726-896fd3fd-6bac-4f77-beeb-4b6a3c51fe92.png)

**Comment by thucke** [comment]
Update:
After going deeping into analyzing prerequisites I've installed kb5020030 following removment and reinstallation now of WSL 1.0.3.
Unfortunately without any change in behaviour.

Calling WSL.EXE as admin results in an error;
![grafik](https://user-images.githubusercontent.com/15112795/215799004-f3ad808f-91bf-446d-9ccc-3961584e3de2.png)

Calling it as a regular user output is as usual.
Especially wsl --version still shown the help page.

**Comment by sbradnick** [comment]
It really seems like you have some remnant of a previous install hanging around (I'm also curious what you have for "Windows Features"). Additionally, if you're "stuck" on a Win10 release < 22H2, these new releases apparently aren't going to work at all. I've progressively updated mine as they came out, being sure I was on the latest Windows release before installing them; I know that's not a possibility for everyone all the time.

I'd suggest REMOVING anything installed already (and doing what you can to verify it's gone), then going to https://github.com/microsoft/wsl/releases and grabbing ones that are <= v1.0.0 and see if any of them behave, > 1.0.0 doesn't seem like a possibility for Win10 21H2. I'm wondering if that (seemingly) old kernel version comes from some 21H2-esque WSL release and you've been attempting installs from github that just aren't compatible w/ Win10 21H2.

But in reality, the biggest roadblock for you seems to be a lack of 22H2 - since I have no issues w/ `wsl --status` or `wsl --version` or TW in general w/ systemd enabled on an up-to-date Win10 install. And I wouldn't run any WSL commands myself as Admin, shouldn't be necessary or a good compare point.

![2023-01-31-130218_1600x967_scrot](https://user-images.githubusercontent.com/84082961/215845826-ae318170-d085-48ab-b3c1-b2479c17eb32.png)

**Comment by thucke** [comment]
Hi @sbradnick,
interesting output trying to install an older version after cleaning up:
![grafik](https://user-images.githubusercontent.com/15112795/216016813-d0cff67c-61bd-4612-96e1-5d698d15c476.png)

It seems that we have to figure out what which KB-Patches have to be installed to get at least W10 version 10.0.22000.0

**Comment by thucke** [comment]
I had success installing version 0.70.0.0 and 0.70.8.0 but still without a change of kernel version and wsl.exe options.
Both installations wre launched using "-ForceUpdateFromAnyVersion  -ForceApplicationShutdown".
I'm running out of ideas.

I cannot find information to the error referenced windows version. According to https://support.microsoft.com/en-us/topic/january-19-2023-kb5019275-os-builds-19042-2546-19044-2546-and-19045-2546-preview-8ae1b678-d38c-4249-848d-06b722e7c0ad the latest 22H2 build is 19044.2546. Numbers starting with 22xxxx seem do identify windows 11 builds.

**Comment by ozdang** [comment, marked as answer]
If there is no version info with `wsl --version` after WSL package install,
I recommend to install latest Cumulative update such as KB5030211 from [Windows Update Catalog](https://www.catalog.update.microsoft.com/Search.aspx?q=Cumulative).
(*be careful to Windows version and machine type)
Also, you can search wsl.exe executable file in your system.
In my case,  wsl package 1.2.5 version is installed in `C:\Program Files\WindowsApps\MicrosoftCorporationII.WindowsSubsystemForLinux_1.2.5.0_x64__8wekyb3d8bbwe\wsl.exe`, but wsl.exe under System32 was not replaced.
After updating KB5030211, it is replaced properly and show `wsl --version`.

**Reply by thucke** [reply]
Hi @ozdang,

KB5030211 did the trick.
after this one had been applied I was able to install the newer WSL from Github.

Thanks a lot
